external/fv3fit/tests/reservoir/test_transformer.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import pytest
+
+from fv3fit.reservoir.transformers.transformer import (
+    encode_columns,
+    DoNothingAutoencoder,
+    decode_columns,
+)
+
+
+@pytest.mark.parametrize("nz, nvars", [(2, 2), (2, 1), (1, 2), (1, 1)])
+def test_DoNothingAutoencoder(nz, nvars):
+    nx = 5
+    transformer = DoNothingAutoencoder(nz)
+    data = [np.ones((nx, nz)) for var in range(nvars)]
+    transformer.encode(data)
+    assert transformer.original_feature_sizes == [nz for var in range(nvars)]
+    encoded_data = np.ones(nz * nvars)
+    assert len(transformer.decode(encoded_data)) == len(data)
+
+
+@pytest.mark.parametrize(
+    "nt, nx, ny, nz, nvars",
+    [(20, 4, 4, 3, 2), (None, 2, 2, 1, 1), (None, 2, 2, None, 1)],
+)
+def test_encode_columns(nt, nx, ny, nz, nvars):
+    shape = tuple([y for y in [nt, nx, ny, nz] if y is not None])
+    expected_shape = (*shape[:-1], nz * nvars) if nz is not None else shape
+    transformer = DoNothingAutoencoder(nz)
+    data_arrs = [np.random.rand(*shape) for var in range(nvars)]
+
+    encoded = encode_columns(data_arrs, transformer=transformer)
+
+    assert encoded.shape == expected_shape
+
+
+@pytest.mark.parametrize(
+    "nx, ny, nz, nvars", [(4, 4, 3, 2), (2, 2, 1, 1), (2, 2, 1, 2)]
```

Comment:
What's the purpose of the last test vs. the first test parameter set?

Reply:
I wanted to test the case of nz=1 with both nvars=1 and >1.